modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsConfigRetriever.scala

```diff
@@ -192,6 +195,75 @@ object TcsConfigRetriever {
+    wfs <- getStatusVal(Nested(active).map{if(_) GuiderSensorOn else GuiderSensorOff}.value, "NGS3 Active status")
+  } yield GuiderConfig(calcProbeTrackingConfig(fol, trk), wfs)
+
+  private def getOdgw1(g: VirtualGemsTelescope, active: IO[Option[Boolean]])
+  : IO[GuiderConfig] = for {
+    prk <- getStatusVal(TcsEpics.instance.odgw1Parked, "ODGW1 parked state")
+    trk <- getStatusVal(getNodChopTrackingConfig(TcsEpics.instance.gemsGuideConfig(g)), "ODGW1 tracking configuration")
+    fol <- getStatusVal(Nested(TcsEpics.instance.odgw1Follow)
+      .map{ if(_) FollowOption.FollowOn else FollowOption.FollowOff}.value, "ODGW1 follow state")
+    wfs <- getStatusVal(Nested(active).map{if(_) GuiderSensorOn else GuiderSensorOff}.value, "ODGW1 Active status")
+  } yield GuiderConfig(prk.fold(ProbeTrackingConfig.Parked, calcProbeTrackingConfig(fol, trk)), wfs)
+
+  private def getOdgw2(g: VirtualGemsTelescope, active: IO[Option[Boolean]])
+  : IO[GuiderConfig] = for {
+    prk <- getStatusVal(TcsEpics.instance.odgw2Parked, "ODGW2 parked state")
+    trk <- getStatusVal(getNodChopTrackingConfig(TcsEpics.instance.gemsGuideConfig(g)), "ODGW2 tracking configuration")
+    fol <- getStatusVal(Nested(TcsEpics.instance.odgw2Follow)
+      .map{ if(_) FollowOption.FollowOn else FollowOption.FollowOff}.value, "ODGW2 follow state")
+    wfs <- getStatusVal(Nested(active).map{if(_) GuiderSensorOn else GuiderSensorOff}.value, "ODGW2 Active status")
+  } yield GuiderConfig(prk.fold(ProbeTrackingConfig.Parked, calcProbeTrackingConfig(fol, trk)), wfs)
+
+  private def getOdgw3(g: VirtualGemsTelescope, active: IO[Option[Boolean]])
+  : IO[GuiderConfig] = for {
+    prk <- getStatusVal(TcsEpics.instance.odgw3Parked, "ODGW3 parked state")
+    trk <- getStatusVal(getNodChopTrackingConfig(TcsEpics.instance.gemsGuideConfig(g)), "ODGW3 tracking configuration")
+    fol <- getStatusVal(Nested(TcsEpics.instance.odgw3Follow)
+      .map{ if(_) FollowOption.FollowOn else FollowOption.FollowOff}.value, "ODGW3 follow state")
+    wfs <- getStatusVal(Nested(active).map{if(_) GuiderSensorOn else GuiderSensorOff}.value, "ODGW3 Active status")
+  } yield GuiderConfig(prk.fold(ProbeTrackingConfig.Parked, calcProbeTrackingConfig(fol, trk)), wfs)
+
+  private def getOdgw4(g: VirtualGemsTelescope, active: IO[Option[Boolean]])
+  : IO[GuiderConfig] = for {
+    prk <- getStatusVal(TcsEpics.instance.odgw4Parked, "ODGW4 parked state")
+    trk <- getStatusVal(getNodChopTrackingConfig(TcsEpics.instance.gemsGuideConfig(g)), "ODGW4 tracking configuration")
+    fol <- getStatusVal(Nested(TcsEpics.instance.odgw4Follow)
+      .map{ if(_) FollowOption.FollowOn else FollowOption.FollowOff}.value, "ODGW4 follow state")
+    wfs <- getStatusVal(Nested(active).map{if(_) GuiderSensorOn else GuiderSensorOff}.value, "ODGW4 Active status")
+  } yield GuiderConfig(prk.fold(ProbeTrackingConfig.Parked, calcProbeTrackingConfig(fol, trk)), wfs)
+
```

Comment:
I could definitely factor out some code in the getNgsX and getOdgwX functions

Reply:
Agree. Is there any reason not to do that now?  It looks like you just need 3 parameters:
* parked
* follow
* guider name